src/Features/Core/Portable/ConvertLinq/AbstractConvertLinqQueryToLinqMethodProvider.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.ConvertLinq
+{
+    internal abstract class AbstractConvertLinqQueryToLinqMethodProvider : AbstractConvertLinqProvider
+    {
+        protected abstract class Analyzer<TSource, TDestination> : AnalyzerBase<TSource, TDestination>
+            where TSource : SyntaxNode
+            where TDestination : SyntaxNode
+        {
+            public Analyzer(SemanticModel semanticModel, CancellationToken cancellationToken) : base(semanticModel, cancellationToken) { }
+
+            protected ImmutableArray<IAnonymousFunctionOperation> FindAnonymousFunctionsFromParentInvocationOperation(SyntaxNode node)
+            {
+                var operation = FindParentInvocationOperation(node);
+
+                if (operation != null)
+                {
+                    return ImmutableArray
+                        .CreateRange(operation.Arguments.Where(a => a.Value.Kind == OperationKind.DelegateCreation)
+                        .Select(argumentOperation => (argumentOperation.Value as IDelegateCreationOperation).Target as IAnonymousFunctionOperation));
```

Comment:
>as [](start = 78, length = 2)

Why are we using ```as``` cast to perform a cast that should not fail, please use regular cast in cases like this.